### ISSUE TITLE
Stuck on processing mobi with large files that need to be chunked

### ISSUE THREAD

**Issue by alvinchenn** [ciromattia/kcc#774]
When I input a large .cbz file (1.1 gb) to convert to MOBI/AZW3, KCC hangs indefinitely during the "Processing MOBI" stage. When I try running the source code, the console output displays a "List index out of range" error when the hanging starts. This error doesn't appear when converting the same file to EPUB. 

A workaround I have found is: rename the .cbz extension to .zip->unzipping the resulting file->input the resulting unzipped directory after clicking "Add directory" in KCC.

This appears to be the same error in this issue, but it says it's supposedly fixed: https://github.com/ciromattia/kcc/issues/582

**Comment by alvinchenn** [comment]
Tested with 6.2.2 and 6.3.0, still hangs for both. Here's the file structure:
<img width="766" alt="Screenshot 2024-11-15 at 10 34 35 PM" src="https://github.com/user-attachments/assets/b076ce6d-4197-4ff4-ad0c-04d2275f4ad4">

**Comment by axu2** [comment]
@alvinchenn can you coy paste the full error when you run from source? 

Can you show me what's inside the cbz after unzip

**Comment by alvinchenn** [comment]
```
qt.gui.imageio: libpng warning: iCCP: known incorrect sRGB profile
Preparing source images...
Checking images...
Processing images...
Creating EPUB file...
Creating EPUB file...
Error calling Python override of QThread::run(): Traceback (most recent call last):
  File "/Users/alvinchen/Downloads/kcc-6.2.2/kindlecomicconverter/KCC_gui.py", line 366, in run
    item, comic2ebook.options.covers[outputPath.index(item)][1]))
          ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^
IndexError: list index out of range
```
<img width="771" alt="Screenshot 2024-11-16 at 11 01 10 PM" src="https://github.com/user-attachments/assets/1a683af2-23d7-4c96-803e-8bea182e37cd">
<img width="539" alt="Screenshot 2024-11-16 at 11 01 53 PM" src="https://github.com/user-attachments/assets/f0efe433-7b0c-4030-9f6f-ffa67104d742">

**Comment by Leonard-Andrei-Dobre-Luxoft** [comment]
Encountering the same issue. I have a 1.5 gig .cbz which is just a folder with all the .jpg files inside.

**Comment by Leonard-Andrei-Dobre-Luxoft** [comment]
Update: Extracting the .cbz with something like 7-zip then just processing the entire folder with the .jpg files works fine.

**Comment by maestrothoughts** [comment]
Yep, I mentioned a similar workaround in my original post. Since it's a trivial workaround, it's a minor issue, but hanging when processing large .cbz files is still a bug.

**Comment by alvinchenn** [comment]
Yep, I mentioned a similar workaround in my original post. Since it's a trivial workaround, it's a minor issue, but hanging when processing large .cbz files is still a bug.

**Comment by axu2** [comment]
@alvinchenn @Leonard-Andrei-Dobre-Luxoft Can you test the above PR?

**Comment by axu2** [comment]
Built on the logic further with 

- #820

**Comment by axu2** [comment]
released in 7.2.0
